### ISSUE TITLE
Add support for Ubuntu 22.04

### ISSUE THREAD

**Issue by manics** [jupyterhub/the-littlest-jupyterhub#813]
### Proposed change
Add support for Ubuntu 22.04

### Alternative options
Do nothing

### Who would use this feature?
People using the latest Ubuntu LTS

### (Optional): Suggest a solution
22.04 doesn't work according to this report: https://discourse.jupyter.org/t/tljh-failing-on-ubuntu-22-04-lts/14097

20.04 seems to work but is not formally documented, see https://github.com/jupyterhub/the-littlest-jupyterhub/issues/720
As part of adding 22.04 we may drop support for 18.04 (as mentioned in https://github.com/jupyterhub/the-littlest-jupyterhub/issues/720)

We should probably also add proper releases as described in https://github.com/jupyterhub/the-littlest-jupyterhub/issues/724 if supporting 22.04 requires non-trivial changes.

**Comment by MridulS** [comment]
22.04 seems to work too without any changes! I just tried out with a clean setup on Ubuntu 22.04 LTS on digital ocean and I was able to start up JupyterHub without any issues.

It's possible there is something specific about the image of Ubuntu 22.04 being used by the user on discourse?

**Comment by wjcapehart** [comment]
My system was an in-place upgrade from 20.04 LTS to 22.04 LTS, not a clean wipe and reinstall.  It is my understanding that there is no "simple or safe" way to remove all of tljh. Am I mistaken?

**Comment by MridulS** [comment]
> It is my understanding that there is no "simple or safe" way to remove all of tljh. Am I mistaken?

Yeah I'm not sure if there is a recommended way of doing this. I would do something like backup the database + user data and then start from scratch while upgrading the ubuntu distribution.

**Comment by wjcapehart** [comment]
I was finally able to get it to work by tar-balling the /home/jupyter-<username> existing directories, deleting /opt/tljh and from there reinstalling.

**Comment by jgiret** [comment]
I made a fresh install on a Ubuntu 22.04 and it seems to work as well.

**Comment by AndrewCRichards** [comment]
Maybe not the best place to post this but hoping it helps others getting "ModuleNotFoundError: No module named 'pip'" when upgrading tljh. In our case we'd just upgraded our server to Ubuntu 24.04LTS. Running bootstrap.py as per the tljh upgrade instructions gave this error. Noting that tljh has its own copies & symlinks of Python and pip, I removed those (the various pips and pythons in /opt/tljh/hub/bin/) (or put them in a backup dir) and reran bootstrap.py which then worked (it found the current versions of python and pip).
I suspect that with the Ubuntu upgrade the local copies & symlinks of python had become broken; removing them allowed bootstrap.py to rebuild them to match the new system, which does indeed have a newer Python version.
